samples/cards/src/index.ts

```diff
@@ -0,0 +1,401 @@
+      break;
+    case '!mixed-action':
+      await send(createActionCardMixed());
+      break;
+    case '!json':
+      await send(createJsonCard());
+      break;
+    case '!profile':
+      await send(editProfileCard());
+      break;
+    case '!profile-input-validation':
+      await send(createProfileCardInputValidation());
+      break;
+    default:
+      await send(
+        new Card().withBody(
+          new TextBlock('Available commands:', { weight: 'bolder' }),
+          new TextBlock('!basic - Show basic card with toggle'),
+          new TextBlock('!form - Show form with multiple inputs'),
+          new TextBlock('!task - Show task management card'),
+          new TextBlock('!actions - Show card with multiple action types')
+        )
+      );
+  }
+});
+
+// :snippet-start: message-handler
+app.on('card.action', async ({ activity, send }) => {
+  const data = activity.value?.action?.data;
+  if (!data?.action) {
+    return {
+      statusCode: 400,
+      type: 'application/vnd.microsoft.error',
+      value: {
+        code: 'BadRequest',
+        message: 'No action specified',
+        innerHttpError: {
+          statusCode: 400,
+          body: { error: 'No action specified' },
+        },
+      },
+    } satisfies AdaptiveCardActionErrorResponse;
+  }
+
+  console.debug('Received action data:', data);
+
+  switch (data.action) {
+    // :remove-start:
+    case 'submit_basic':
+      await send(`Notification preference set to: ${data.notify}`);
+      break;
+
+    case 'submit_form':
+      await send(
+        `Form submitted!\nName: ${data.name}\nComments: ${data.comments}\nColor: ${data.color}`
+      );
+      break;
+
+    case 'create_task':
+      await send(
+        `Task created!\nTitle: ${data.title}\nDescription: ${data.description}\nPriority: ${data.priority}\nDue Date: ${data.due_date}`
+      );
+      break;
+
+    // :remove-end:
+    case 'submit_feedback':
+      await send(`Feedback received: ${data.feedback}`);
+      break;
+
+    case 'purchase_item':
+      await send(`Purchase request received for game: ${data.choiceGameSingle}`);
+      break;
+
+    case 'save_profile':
+      await send(
+        `Profile saved!\nName: ${data.name}\nEmail: ${data.email}\nSubscribed: ${data.subscribe}`
+      );
+      break;
+
+    default:
+      return {
+        statusCode: 400,
+        type: 'application/vnd.microsoft.error',
+        value: {
+          code: 'BadRequest',
+          message: 'Unknown action',
+          innerHttpError: {
+            statusCode: 400,
+            body: { error: 'Unknown action' },
+          },
+        },
+      } satisfies AdaptiveCardActionErrorResponse;
+  }
+
+  return {
+    statusCode: 200,
+    type: 'application/vnd.microsoft.activity.message',
+    value: 'Action processed successfully',
+  } satisfies AdaptiveCardActionMessageResponse;
+});
+// :snippet-end:
+
+(async () => {
+  await app.start(+(process.env.PORT || 3000));
+})();
```

Comment:
Since this file is over 250 lines of code, do we want to extract out some of the helper methods for readability?

Reply:
We could, but it's kinda nice that all the stuff for this very specific use-case are all together. I could go either way. What do you think?

Reply:
Hmm I feel the same. I try to keep smaller files in general because I know that's @aacebo's preference, but I understand your POV as well. @aacebo tie-breaker?